zilliqa/src/consensus.rs

```diff
@@ -351,6 +355,7 @@ impl Consensus {
             state,
             db,
             new_transactions_priority: BTreeMap::new(),
+            rng: <rand_chacha::ChaCha8Rng as rand_core::SeedableRng>::seed_from_u64(0),
```

Comment:
I worry that we'll use this rng to generate something sensitive down the line and regret having a very predictable seed. Seeding it from the secret key sounds like a good idea to me. Can we do that instead?

Also this should probably be `ChaCha20Rng` for the same reason?

Reply:
Sorry I didn't read this comment before making my suggestion. I was basing the suggestion of 20 rounds on:
> With the ChaCha algorithm it is possible to choose the number of rounds the core algorithm should run. The number of rounds is a tradeoff between performance and security, where 8 rounds is the minimum potentially secure configuration, and 20 rounds is widely used as a conservative choice.

from https://docs.rs/rand_chacha/latest/rand_chacha/struct.ChaCha8Rng.html

Reply:
Ah sorry I didn't register the public key point. I don't think using the public key is sensible if our argument is that we want these random numbers to be unpredictable. What's wrong with using the private key?